uaclient/api/api.py

```diff
@@ -0,0 +1,55 @@
+from importlib import import_module
+from typing import Callable, Optional
+
+from uaclient.api.data_types import APIData, APIResponse
+from uaclient.api.utils import VALID_ENDPOINTS, error_out
+from uaclient.data_types import DataObject
+
+
+def call_api(args, *, cfg):
+    if args.endpoint_path not in VALID_ENDPOINTS:
+        return error_out(
+            APIError("{} is not a valid endpoint".format(args.endpoint_path))
+        )
+
+    module = import_module("uaclient.api." + args.endpoint_path)
+    endpoint = module.endpoint
+
+    try:
+        if endpoint.options_cls:
+            options = endpoint.options_cls(**args.options)
```

Comment:
I think we may need to do something like tell users to run
```
ua api u.pro.endpoint.v7 key=value key2=value2
```
with those `=` being necessary.

Then here we would run something like:
```suggestion
            kwargs = dict([pair.split("=") for pair in args.options])
            options = endpoint.options_cls(**kwargs)
```
probably with better error handling though